include/api.php

```diff
@@ -3919,7 +3919,9 @@ function api_friendica_direct_messages_search($type){
 
 		$profile_url = $user_info["url"];
 		// message if nothing was found
-		if (count($r) == 0) 
+		if (!dbm::is_result($r))
```

Comment:
Inalid database result (means failed query) or zero rows found (but valid query) is different. Here I added this to differenciate with both.